### PR TITLE
James 1779 document download endpoint limitations

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

**Comment by mbaechler** [comment]
test this please

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadEndpoint.feature [hunk 1]
```diff
@@ -4,27 +4,49 @@ Feature: Download endpoint
...
+    Then the user should receive a not found response
+
+  @Ignore
+  Scenario: A user should not have access to someone else attachment
+    Given "userb@domain.tld" is connected
+    When "userb@domain.tld" downloads "a1"
+    Then the user should receive a not found response
+
+  @Ignore
+  Scenario: A user should not have access to someone else attachment
```
Comment (aduprat) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadEndpoint.feature]: Same scenario as the previous

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/UserStepdefs.java [hunk 2]
```diff
@@ -19,36 +19,97 @@
...
 
 @ScenarioScoped
 public class UserStepdefs {
 
     private final MainStepdefs mainStepdefs;
-    protected String username;
-    protected AccessToken accessToken;
-
+    
+    protected Map<String, String> passwordByUsers;
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/UserStepdefs.java]: ByUser?

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java [hunk 3]
```diff
@@ -60,6 +60,7 @@
 import com.jayway.awaitility.Duration;
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java]: Why don't you do it every where in jmap integration testing ?

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/UserStepdefs.java [hunk 4]
```diff
@@ -19,36 +19,97 @@
...
     @Given("^a domain named \"([^\"]*)\"$")
     public void createDomain(String domain) throws Throwable {
         mainStepdefs.jmapServer.serverProbe().addDomain(domain);
+        domains.add(domain);
     }
 
-    @Given("^a current user with username \"([^\"]*)\" and password \"([^\"]*)\"$")
-    public void createUserWithPasswordAndAuthenticate(String username, String password) throws Throwable {
-        this.username = username;
+    @Given("^some users (.*)$")
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/UserStepdefs.java]: see https://github.com/cucumber/cucumber-jvm/pull/371/commits/5ebb21a6882cf84810fc46381655db73501ca8a3#diff-f7069067d4a61821737af3ed22c09142R20 maybe

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java [hunk 5]
```diff
@@ -88,6 +89,10 @@ public void setup() throws Throwable {
         jmapServer.start();
         RestAssured.port = jmapServer.getJmapPort();
         RestAssured.config = newConfig().encoderConfig(encoderConfig().defaultContentCharset(Charsets.UTF_8));
+        RestAssured.requestSpecification = new RequestSpecBuilder()
+        		.setContentType(ContentType.JSON)
+        		.setAccept(ContentType.JSON)
+        		.build();
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java]: This should be an other commit now? Seems to be not related at all to the commit message.

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/UserStepdefs.java (hunk 4 above)
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/UserStepdefs.java]: already seen but : 
x the default delimiter is "\s," anyway
x it won't handle quoted string anyway and thus add nothing

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java (hunk 5 above)
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java]: Yes, I'll make the change on all files concerning the content type and accept in a preparation step commit

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadEndpoint.feature (hunk 1 above)
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadEndpoint.feature]: true

**Comment by mbaechler** [comment]
sorry for the push -f by it was needed to split the removal of JSON everywhere from the remaining code

File: mpt/impl/smtp/core/src/main/java/org/apache/james/mpt/smtp/ForwardSmtpTest.java [hunk 6]
```diff
@@ -75,15 +76,17 @@ public void setUp() throws Exception {
...
+        		.setContentType(ContentType.JSON)
+        		.setAccept(ContentType.JSON)
+        		.build();
     }
 
     @Test
     public void forwardingAnEmailShouldWork() throws Exception {
         scriptTest("helo", Locale.US);
 
         given()
```
Comment (chibenwa) [inline mpt/impl/smtp/core/src/main/java/org/apache/james/mpt/smtp/ForwardSmtpTest.java]: Maybe here you can even remove the given() that do not add anything semantically...

**Comment by aduprat** [comment]
:+1:

**Comment by chibenwa** [comment]
:+1:

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/UserStepdefs.java (hunk 4 above)
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/UserStepdefs.java]: then we could use tabs, but not necessary for this one.

**Comment by rouazana** [comment]
:+1:

**Comment by aduprat** [comment]
merged
